xtask/src/main.rs

```diff
@@ -913,3 +922,156 @@ fn run_doc_tests(workspace: &Path, args: ExampleArgs) -> Result<()> {
+
+    // Check doc-tests
+    run_doc_tests(
+        workspace,
+        ExampleArgs {
+            package: Package::EspHal,
+            chip: args.chip,
+            example: None,
+            debug: true,
+        },
+    )
+    .inspect_err(|_| failure = true)
+    .ok();
+
+    // Check documentation
+    build_documentation(
+        workspace,
+        BuildDocumentationArgs {
+            packages: vec![Package::EspHal, Package::EspWifi, Package::EspHalEmbassy],
+            chips: vec![args.chip],
+            base_url: None,
+        },
+    )
+    .inspect_err(|_| failure = true)
+    .ok();
+
+    // for chips with esp-lp-hal: Build all supported examples for the low-power
+    // core first
+    if args.chip.has_lp_core() {
+        // Build prerequisite examples (esp-lp-hal)
+        // `examples` copies the examples to a folder with the chip name as the last
+        // path element then we copy it to the place where the HP core example
+        // expects it
+        examples(
+            workspace,
+            ExampleArgs {
+                package: Package::EspLpHal,
+                chip: args.chip,
+                example: None,
+                debug: false,
+            },
+            CargoAction::Build(PathBuf::from(format!(
+                "./esp-lp-hal/target/{}/release/examples",
+                args.chip.target()
+            ))),
+        )
+        .inspect_err(|_| failure = true)
+        .and_then(|_| {
+            let from_dir = PathBuf::from(format!(
+                "./esp-lp-hal/target/{}/release/examples/{}",
+                args.chip.target(),
+                args.chip.to_string()
+            ));
+            let to_dir = PathBuf::from(format!(
+                "./esp-lp-hal/target/{}/release/examples",
+                args.chip.target()
+            ));
+            from_dir.read_dir()?.for_each(|entry| {
+                let entry = entry.unwrap();
+                let path = entry.path();
+                let to = to_dir.join(entry.file_name());
+                fs::copy(path, to).expect("Failed to copy file");
+            });
+            Ok(())
+        })
+        .ok();
+
+        // Check documentation
+        build_documentation(
+            workspace,
+            BuildDocumentationArgs {
+                packages: vec![Package::EspLpHal],
+                chips: vec![args.chip],
+                base_url: None,
+            },
+        )
+        .inspect_err(|_| failure = true)
+        .ok();
+    }
+
+    // Make sure we're able to build the HAL without the default features enabled
+    build_package(
+        workspace,
+        BuildPackageArgs {
+            package: Package::EspHal,
+            target: Some(args.chip.target().to_string()),
+            features: vec![args.chip.to_string()],
+            toolchain: if args.chip.is_xtensa() {
+                Some(String::from("esp"))
+            } else {
+                Some(String::from("stable"))
+            },
```

Comment:
IIRC, we have this check in a few places, maybe we could do `args.chip.toolchain()` and move this check inside the method.

Reply:
I guess we sometimes want/need stable and sometimes nightly - probably we can have some default and add an option to override it

Reply:
just checked - this is the only place where we explicitly set "stable" (otherwise we use whatever the default is for RV) - the only reason I added this was the CI failing w/o it - probably better to check what is/was wrong with the CI job instead of working around that here

Thanks for the comment here (which reminded me of that)